components/gitpod-protocol/src/util/tracing.ts

```diff
@@ -22,9 +22,19 @@ export type TraceContextWithSpan = TraceContext & {
 export namespace TraceContext {
     export function startSpan(operation: string, parentCtx?: TraceContext): opentracing.Span {
         const options: opentracing.SpanOptions = {};
-        if (parentCtx && parentCtx.span && !!parentCtx.span.context().toSpanId()) {
-            options.childOf = parentCtx.span;
+
+        // try out: https://github.com/jaegertracing/jaeger-client-node/issues/432
+        if (!!parentCtx?.span) {
+            const ctx = parentCtx?.span?.context();
+            if (ctx && !!ctx.toTraceId()) {
```

Comment:
`traceId` was missing here as well.

Reply:
wait, maybe not here 🤔

Reply:
Also we had a check for `!!parentCtx.span.context().toSpanId()` before, I think we should keep it in as well.

Reply:
done